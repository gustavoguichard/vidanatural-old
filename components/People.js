import { useState, useRef, memo, useEffect } from 'react'
import Link from 'next/link'
import take from 'lodash/take'
import shuffle from 'lodash/shuffle'
import { Button } from 'react-bulma-components'
import { FaPlus } from 'react-icons/fa'

import Testimonial from 'components/testimonials/Testimonial'
import Slogan from 'components/Slogan'
import Masonry from 'components/Masonry'
import { useWindowDimensions, useProcessOnce, useMedia } from 'utils/hooks'

import testimonials from 'content/testimonials'
import 'styles/banner.scss'
import 'styles/people.scss'

const Intro = memo(({ isSidebar, children }) => (
  <div
    className={
      isSidebar ? 'tile is-vertical content-wrapper' : 'testimonial-item'
    }
    style={isSidebar ? { order: 2 } : {}}
  >
    <div
      className={isSidebar ? 'banner-content-wrapper' : 'tile-content'}
      style={isSidebar ? { justifyContent: 'center', maxHeight: '100vh' } : {}}
    >
      <div className={isSidebar ? 'banner-content' : 'inner-content'}>
        <Slogan responsive />
        <p style={{ margin: '2em 0' }}>
          Temos um profundo respeito pelos nossos consumidores e isso nos
          motivou a criar este espaço para todos eles no nosso site.
          <br />
          Confira aqui o que realmente motivas as pessoas a usarem os cosméticos
          da <b>Vida Natural</b>.
        </p>
        <p style={{ margin: '2em 0' }}>
          Vale lembrar, que este espaço também é seu.
          <br />
          Gostaria de nos contar algo sobre a sua experiência com algum produto
          da nossa linha de cosméticos?
          <br />
          <Link href="#contato">
            <a>Converse com a gente!</a>
          </Link>{' '}
          Estamos prontos e curiosos para ouvir.
        </p>
      </div>
      {children}
    </div>
  </div>
))

const People = () => {
  const [faceCount, setFaceCount] = useState(6)
  const [isOpen, setIsOpen] = useState(false)
  const [wrapperWidth, setWrapperWidth] = useState(0)
  const shuffledTestimonials = useProcessOnce(shuffle, testimonials)
  const testimonialsToShow = take(shuffledTestimonials, faceCount)

  const { width } = useWindowDimensions()
  const wrapper = useRef(null)
  useEffect(
    () => setWrapperWidth(wrapper.current.getBoundingClientRect().width),
    [width, isOpen],
  )
  const columns = wrapperWidth ? Math.round(wrapperWidth / 365) : 2
  const isDesktop = useMedia('desktop')

  const isShowingAll = faceCount >= testimonials.length
  const loadMoreFaces = () => setFaceCount(faceCount + 4)
  const openContent = () => {
    setIsOpen(true)
    loadMoreFaces()
  }

  const showSidebar = isDesktop && !isOpen
  const showMoreTile = !(showSidebar || isShowingAll)

  return (
    <div id="eu-uso">
      <div className="tile">
        {showSidebar && (
          <Intro isSidebar>
            <div className="banner-content">
              <Button
                color="light"
                className="is-large"
                rounded
                outlined
                onClick={openContent}
              >
                + Depoimentos
              </Button>
            </div>
          </Intro>
        )}
        <div className="tile" ref={wrapper}>
          <Masonry columns={columns}>
            {isDesktop || <Intro />}
            {testimonialsToShow.map((testimonial, index) => (
              <Testimonial key={index} {...testimonial} />
            ))}
            {showMoreTile && (
              <div className="testimonial-item">
                <div className="tile-content center" onClick={loadMoreFaces}>
                  <p>Ver mais depoimentos</p>
                  <button
                    title="Mais depoimentos"
                    className="plus-bt"
                    onClick={loadMoreFaces}
                  >
                    <FaPlus />
                  </button>
                </div>
              </div>
            )}
          </Masonry>
        </div>
      </div>
    </div>
  )
}

export default memo(People)
