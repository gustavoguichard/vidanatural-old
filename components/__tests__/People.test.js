import { render } from 'react-testing-library'
import People from 'components/People'

it('renders correctly', () => {
  const { container } = render(<People router={{ pathname: '/' }} />)
  expect(container).toMatchSnapshot()
})
