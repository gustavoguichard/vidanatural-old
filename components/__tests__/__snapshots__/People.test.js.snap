// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    id="eu-uso"
  >
    <div
      class="tile"
    >
      <div
        class="tile"
      >
        <div
          class="jsx-3973384840 masonry-tile"
        >
          <div
            class="jsx-3973384840 vert-tile"
          >
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Cristina Scudeler
              </h3>
              <div>
                <p>
                  Há mais de dois anos que uso o desodorante em pasta em 
                  <strong>
                    não troco
                  </strong>
                   por outro. 
                  <strong>
                    Super eficiente
                  </strong>
                  , 
                  <strong>
                    100% Natural
                  </strong>
                  , e rende muito, uso por pelo menos 
                  <strong>
                    2 meses o mesmo
                  </strong>
                   tubinho.
                  <br />
                    Super feliz por esse produto existir!
                </p>
              </div>
              <em>
                Empresária - São Paulo / SP
              </em>
            </article>
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Juliana Eichenberg
              </h3>
              <div>
                <p>
                  A Vida Natural, mais do que uma marca, é a forma de vida da sua querida idealizadora. Não há limites separando a vida da Lila da Vida Natural e a Vida Natural da vida da Lila.
                  <br />
                    Essa fusão, da criadora e da criação, é que torna os 
                  <strong>
                    produtos que uso tão especiais
                  </strong>
                   para mim. Tudo nasce de uma 
                  <strong>
                    verdade profunda
                  </strong>
                  , de um anseio por 
                  <strong>
                    sustentabilidade, regeneração e saúde
                  </strong>
                  . O que é evidente e palpável na postura da Lila e nos benefícios que sentimos ao nos nutrir através da Vida Natural.
                  <br />
                    Obrigada, Vida Natural! Obrigada, Lila! Muito bom trilhar esse caminho com vocês, um grande privilégio!
                </p>
              </div>
              <em>
                Professora de Yoga - Garopaba / SC
              </em>
            </article>
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Samira Abdalah
              </h3>
              <div>
                <p>
                  Usar os produtos da Vida Natural é 
                  <strong>
                    se sentir em harmonia com o corpo
                  </strong>
                   e o 
                  <strong>
                    meio ambiente
                  </strong>
                  . São produtos feitos com toda 
                  <strong>
                    dedicação e carinho
                  </strong>
                   e  de muita qualidade. Super ultra recomendado!
                </p>
              </div>
              <em>
                Mestra em Educação - Montenegro / RS
              </em>
            </article>
            <div
              class="testimonial-item"
            >
              <div
                class="tile-content center"
              >
                <p>
                  Ver mais depoimentos
                </p>
                <button
                  class="plus-bt"
                  title="Mais depoimentos"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="jsx-3973384840 vert-tile"
          >
            <div
              class="testimonial-item"
            >
              <div
                class="tile-content"
              >
                <div
                  class="inner-content"
                >
                  <div
                    class="slogan"
                    style="display: block; background: url(/static/slogan.png) no-repeat center; background-size: contain; height: 45px; overflow: hidden; text-indent: -999em;"
                  >
                    <h2>
                      Eu uso
                    </h2>
                    <h3>
                      Cosmética Consciente
                    </h3>
                  </div>
                  <p
                    style="margin: 2em 0px;"
                  >
                    Descubra o que motiva as pessoas a usarem os nossos cosméticos - Vida Natural
                  </p>
                </div>
              </div>
            </div>
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Maia Silva
              </h3>
              <div>
                <p>
                  Eu amo escolher 
                  <strong>
                    diariamente
                  </strong>
                   a Vida Natural para 
                  <strong>
                    nutrir
                  </strong>
                   meu corpo. Falar de cosméticos é falar de prevenção e alimentação também.
                  <br />
                    Além de serem uma 
                  <strong>
                    escolha de consumo consciente e responsável
                  </strong>
                  , eles deixam minha pele cheirosa e nutrida por muito tempo, 
                  <strong>
                    livre de toxinas
                  </strong>
                   que ao longo dos anos possam danificar minha 
                  <strong>
                    saúde
                  </strong>
                  .
                </p>
              </div>
              <em>
                Psicóloga Social - Argentina
              </em>
            </article>
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Thais e Bruno Padoveze
              </h3>
              <div>
                <p>
                  Há 3 anos comprovando a 
                  <strong>
                    eficácia
                  </strong>
                  , a 
                  <strong>
                    pureza
                  </strong>
                   e a riqueza 
                  <strong>
                    natural
                  </strong>
                   de toda a linha de produtos. Muitos, mas muitos amigos hoje em dia também são clientes e fãs dos cosméticos Vida Natural.
                  <br />
                    Ainda bem que existem produtos tão 
                  <strong>
                    seguros e naturais
                  </strong>
                   ao alcance de nossas mãos aqui no Brasil. Obrigado Lila!
                </p>
              </div>
              <em>
                Goiânia
              </em>
            </article>
            <article
              class="testimonial-content"
            >
              <h3
                class="title is-4"
              >
                Maurício Frighetto
              </h3>
              <div>
                <p>
                  Todos os produtos que experimentei da Vida Natural foram 
                  <strong>
                    muito eficientes
                  </strong>
                  . O que já é um grande ponto positivo. Mas talvez o mais importante para mim seja saber que eles foram feitos com 
                  <strong>
                    carinho
                  </strong>
                   e, principalmente, sem o uso de 
                  <strong>
                    nenhum produto de origem animal
                  </strong>
                  . É um grande diferencial. Por estes dois motivos, recomendo fortemente
                </p>
              </div>
              <em>
                Jornalista - Florianópolis / SC
              </em>
            </article>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
